OpenLaTeXStudio/EditorModule/src/latexstudio/editor/menu/OpenTemplate.java

```diff
@@ -0,0 +1,121 @@
+@NbBundle.Messages("CTL_OpenTemplate=Open Template")
+
+public final class OpenTemplate extends JDialog implements ActionListener {
+
+    private static final String pathToTemplatesDirectory = "/latexstudio/editor/resources/templates/";
+
+    private final EditorTopComponent etc = new TopComponentFactory<EditorTopComponent>()
+            .getTopComponent(EditorTopComponent.class.getSimpleName());
+
+    private OpenTemplate() {
+        //Dialog window properties
+        setTitle("Choose Template");
+        setLocationRelativeTo(etc);
+        setLayout(new BorderLayout(10, 10));
+        setResizable(false);
+        setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout(10, 10));
+        setModalityType(ModalityType.APPLICATION_MODAL);
+
+        add(new OpenTemplatePanel(this, getListModelWithTemplates(readXML())), BorderLayout.CENTER);
+        pack();
+    }
+
+    /**
+     @return Array of nodes that contain informations about preinstalled
+     templates
+     */
+    private ArrayList<Element> readXML() {
+        try {
+            Document templatesXML = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(getClass().getResourceAsStream("/latexstudio/editor/resources/templates.xml"));
+
+            Element rootElement = templatesXML.getDocumentElement();
+            NodeList templatesNodeList = rootElement.getChildNodes();
+
+            ArrayList<Element> templates = new ArrayList<Element>();
+            for (int i = 0; i < templatesNodeList.getLength(); i++) {
+                Node node = templatesNodeList.item(i);
+                if (node instanceof Element) {
+                    templates.add((Element) node);
+                }
+            }
+            
+            return templates;
+        } catch (IOException e) {
+        } catch (SAXException e) {
+        } catch (ParserConfigurationException e) {
+        }
+        return null;
+    }
+    
+    /**
+    
+    @param path template filename
+    @return template full path
+    */
+    private String getFullPathToTemplate(String path) {
+        return pathToTemplatesDirectory + path;
+    }
+
+    private DefaultListModel<Template> getListModelWithTemplates(ArrayList<Element> elements) {
+        DefaultListModel<Template> listModel = new DefaultListModel<Template>();
+        for (Element e : elements) {
+            listModel.addElement(getTemplateFromElement(e.getChildNodes()));
+        }
+        return listModel;
+    }
+
+    private Template getTemplateFromElement(NodeList templateNodes) {
+        ArrayList<Element> templateElements = new ArrayList<Element>(4);
+        for (int i = 0; i < templateNodes.getLength(); i++) {
+            Node node = templateNodes.item(i);
+            if (node instanceof Element) {
+                templateElements.add((Element) node);
+            }
+        }
+
+        Text name = (Text) templateElements.get(0).getFirstChild();
```

Comment:
It would be better (safer) to retrieve nodes by their name. That way it will still when we change the order in XML.